src/sage/categories/rings.py

```diff
@@ -888,27 +921,26 @@ def ideal(self, *args, **kwds):
                         break
                     elif isinstance(first, (list, tuple, GeneratorType)):
                         gens = first
+                    elif isinstance(first, Parent) and self.has_coerce_map_from(first):
+                        gens = first.gens()  # we have a ring as argument
```

Comment:
I don't quite understand, the INPUT says that `gens` should be - essentially - generators.  I don't see a doctest where they are rings, either.